airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -172,27 +177,89 @@ def _doc_upload(
+    """Apply any prerelease overrides to the metadata file before uploading it to GCS."""
+    if validator_opts.prerelease_tag is None:
+        return metadata_dict
 
     # replace any dockerImageTag references with the actual tag
     # this includes metadata.data.dockerImageTag, metadata.data.registries[].dockerImageTag
     # where registries is a dictionary of registry name to registry object
-    metadata_dict = to_json_sanitized_dict(metadata, exclude_none=True)
     metadata_dict["data"]["dockerImageTag"] = validator_opts.prerelease_tag
     for registry in get(metadata_dict, "data.registries", {}).values():
         if "dockerImageTag" in registry:
             registry["dockerImageTag"] = validator_opts.prerelease_tag
 
-    # write metadata to yaml file in system tmp folder
-    tmp_metadata_file_path = Path("/tmp") / metadata.data.dockerRepository / validator_opts.prerelease_tag / METADATA_FILE_NAME
-    tmp_metadata_file_path.parent.mkdir(parents=True, exist_ok=True)
-    with open(tmp_metadata_file_path, "w") as f:
-        yaml.dump(metadata_dict, f)
+    return metadata_dict
+
+
+def _get_git_info_for_file(original_metadata_file_path: Path) -> Optional[GitInfo]:
+    """
+    Add additional information to the metadata file before uploading it to GCS.
+
+    e.g. The git commit hash, the date of the commit, the author of the commit, etc.
+
+    """
+    repo = git.Repo(search_parent_directories=True)
+    try:
+        # get the commit hash for the last commit that modified the metadata file
+        commit_sha = repo.git.log("-1", "--format=%H", str(original_metadata_file_path))
+        commit_timestamp = repo.git.log("-1", "--format=%ct", str(original_metadata_file_path))
+        commit_author = repo.git.log("-1", "--format=%an", str(original_metadata_file_path))
+        commit_author_email = repo.git.log("-1", "--format=%ae", str(original_metadata_file_path))
+
+        return GitInfo(
+            commit_author=commit_author,
+            commit_author_email=commit_author_email,
+            commit_timestamp=commit_timestamp,
+            commit_sha=commit_sha,
+        )
+    except git.exc.GitCommandError as e:
+        if "unknown revision or path not in the working tree" in str(e):
+            logging.warning(f"Metadata file {original_metadata_file_path} is not tracked by git, skipping author info attachment.")
+            return None
+        else:
+            raise e
+
+
+def _apply_author_info_to_metadata_file(metadata_dict: dict, original_metadata_file_path: Path) -> dict:
+    """Apply author info to the metadata file before uploading it to GCS."""
+    git_info = _get_git_info_for_file(original_metadata_file_path)
+    if git_info:
+        # Apply to the nested / optional field at metadata.data.ab_internal.git
+        git_info_dict = to_json_sanitized_dict(git_info, exclude_none=True)
+        metadata_dict = set_(metadata_dict, "data.ab_internal.git", git_info_dict)
```

Comment:
nit: shall we use a different namespace than `ab_internal` for *auto-generated* metadata"? What if someone sets the `data.ab_internal.git` manually? It will be silently overwritten, won't it?

Reply:
It will! Ill move to a generated field.

Reply:
Whoops. Realized I missed this

Reply:
Working on it now